extensions/ql-vscode/src/common/vscode/vscode-app.ts

```diff
@@ -69,4 +69,10 @@ export class ExtensionApp implements App {
   public createEventEmitter<T>(): AppEventEmitter<T> {
     return new VSCodeAppEventEmitter<T>();
   }
+
+  public get environment(): EnvironmentContext {
+    return {
+      language: vscode.env.language,
```

Comment:
Should this be an object with a getter? I believe you can change the VS Code language while it's running and this would not reflect that if you're passing `app.environment` to another class.

Reply:
Thanks for pointing that out. I agree using a getting would be good here.